### PR TITLE
Use strong params in Dependencies API controller.

### PR DESCRIPTION
**Pull request by simi**
- it helps to filter out invalid (in attached report probably some SQLi testing) params exploding app

fixes https://app.honeybadger.io/projects/40972/faults/68481662/47fe907c-15ff-11eb-8f2a-9b48b4dba36c#notice-summary

### CODE REVIEW

File: test/functional/api/v1/dependencies_controller_test.rb [hunk 1]
```diff
@@ -31,6 +31,23 @@ class Api::V1::DependenciesControllerTest < ActionController::TestCase
     end
   end
 
+  # INVALID GEMS:
+  context "On GET to index --> with invalid gems --> JSON" do
```
Comment (sonalkr132) [inline test/functional/api/v1/dependencies_controller_test.rb:35]: can we please be more specific, perhaps "with hash in gems params"?

File: test/functional/api/v1/dependencies_controller_test.rb [hunk 2]
```diff
@@ -31,6 +31,23 @@ class Api::V1::DependenciesControllerTest < ActionController::TestCase
...
+    setup do
+      rubygem = create(:rubygem, name: "testgem")
+      @version = create(:version, number: "1.0.0", rubygem_id: rubygem.id)
```
Comment (sonalkr132) [inline test/functional/api/v1/dependencies_controller_test.rb:38]: do we need the above two lines in the setup?

Comment (simi) [reply test/functional/api/v1/dependencies_controller_test.rb:38]: probably not, I just copied tests around doing the same (even not used)

File: test/functional/api/v1/dependencies_controller_test.rb (hunk 1 above)
Comment (simi) [reply test/functional/api/v1/dependencies_controller_test.rb:35]: sure

Comment (simi) [reply test/functional/api/v1/dependencies_controller_test.rb:35]: Would be "malformed" enough?

**Comment by simi** [comment]
@sonalkr132 updated, also checked specs still fail without fix.

**Review by sonalkr132 — APPROVED** [review]

**Comment by sonalkr132** [comment]
Thank you :100:

**Pull request merged**